skimage/filters/_unsharp_mask.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import division
+import numpy as np
+from scipy.ndimage.filters import gaussian_filter
+from skimage.util.dtype import dtype_limits
+from skimage import img_as_float
+
+
+def _unsharp_mask_single_channel(image, radius, amount, vrange):
+    """Single channel implementation of the unsharp masking filter."""
+
+    blurred = gaussian_filter(image,
+                              sigma=radius,
+                              mode='reflect')
+
+    result = image + (image - blurred) * amount
+    if vrange is not None:
+        return np.clip(image, vrange[0], vrange[1], out=result)
```

Comment:
Should be clipping `result`, not `image`.

Reply:
Good catch!